### ISSUE TITLE
multi cursor with comment character in python can be wrong

### ISSUE THREAD

**Issue by WindSoilder** [zed-industries/feedback#569]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Assume that we have the following file:
```python
# main.py

class A:
    def b():
        pass
```

If we put the cursor at the following line(`|` indicate the cursor)
```python
|class A:
|    def b():
|        pass
```
And press `#`, the result will be:
```python
#class A:
#    def b():
    #        pass
```

### Expected behavior

After press `#`, the result should be:
```python
#class A:
#    def b():
#        pass
```

### Environment

Zed 0.59.0 – /Applications/Zed.app \nmacOS 12.6 \narchitecture x86_64

**Comment by JosephTLyons** [comment]
Hey @WindSoilder, are you actually typing in the `#` character or are you using `cmd-/`?

For me, if I place down multiple cursors and manually type `#`, then it works the way you expect, all of the comment symbols are vertically aligned.

If I use place down multiple cursors and use `cmd-/`, then it results in the comment symbols being indented at the same level of the code - this is because each cursor is adding their own comment symbol and each cursor only has the context of the given line, so it uses that lines indentation.  This seems to be consistent with VS Code.

If you are wanting to comment out a block of code, does the workflow of highlighting the code block in one selection and using `cmd-/` work for you?

**Comment by JosephTLyons** [comment]
Ok, it looks like I misunderstood because I am not able to reproduce the issue you are seeing, but @maxbrunsfeld is able to reproduce it, so it seems like this bug is conditional on something, but we don't know what it is yet.

**Comment by WindSoilder** [comment]
> If you are wanting to comment out a block of code, does the workflow of highlighting the code block in one selection and using cmd-/ work for you?

Ah, I'm not using `cmd-/` a lot, I should try it :-)

**Comment by JosephTLyons** [comment]
> > If you are wanting to comment out a block of code, does the workflow of highlighting the code block in one selection and using cmd-/ work for you?
> 
> Ah, I'm not using `cmd-/` a lot, I should try it :-)

Yeah, that's a good technique for commenting out a bunch of code, without having to make multiple cursors.  That being said, we do have a bug here, appreciate you reporting it :)
